CSharpSourceCode/Abilities/SeekerController.cs

```diff
@@ -0,0 +1,48 @@
+﻿using TaleWorlds.Engine;
+using TaleWorlds.Library;
+using TOW_Core.Battle.AI.Decision;
+
+namespace TOW_Core.Abilities.Scripts
+{
+    public class SeekerController
+    {
+        private Target _target;
+        private Vec3 _prevError;
+        private SeekerParameters _parameters;
+        private bool enabled = true;
+
+        public SeekerController(Target target, SeekerParameters parameters)
+        {
+            _target = target;
+            _prevError = Vec3.Zero;
+            _parameters = parameters;
+        }
+
+        public MatrixFrame CalculateRotatedFrame(MatrixFrame globalFrame, float dt)
+        {
+            if (enabled && _target != null && (_target.Agent != null || _target.Formation.CountOfUnits > 0))
+            {
+                var particleDirection = globalFrame.origin + globalFrame.rotation.f.NormalizedCopy();
+                var error = _target.Position + new Vec3(0, 0, 2) - particleDirection;
+                if (error.Length < _parameters.DisableDistance)
+                {
+                    enabled = false;
+                    return globalFrame;
+                }
+
+                if (error.Length < _parameters.MaxDistance && error.Length > _parameters.MinDistance)
+                {
+                    var correction = error * _parameters.Proportional + (error - _prevError) * _parameters.Derivative;
```

Comment:
Potentially the line I need a little more explaination... just from the variable names or the calculation I can't really understand the correction.